### PR TITLE
Jdp230101 12

### PR DESCRIPTION
**Pull request by CodeStudent1742**
Initial Entity implementation of Product. Need to add Relations on later stage

### CODE REVIEW

**Review by piotr-grosicki — CHANGES_REQUESTED** [review]
tak jak pisalem na komunikatorze o relacjach + konflikty w pliku build.gradle (probujesz kolejny raz dodac lomboka

**Comment by CodeStudent1742** [comment]
Product with relations. Cart and Order needed part added

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@Entity(name = "CARTS")
```
Comment (waleczna) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java:17]: `CARTS` should be a table name but not an entity name. It should look like: 
```
@Entity
@Table (name = "CARTS")
```

File: src/main/java/com/kodilla/ecommercee/domain/Product.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import lombok.NoArgsConstructor;
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+import java.util.List;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+@Entity(name = "PRODUCTS")
```
Comment (waleczna) [inline src/main/java/com/kodilla/ecommercee/domain/Product.java:14]: `PRODUCTS` should be a table name but not an entity name. It should look like: 
```
@Entity
@Table (name = "PRODUCTS")
```

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+public class Cart {
+
+    @Id
+    @GeneratedValue
+    @NotNull
+    @Column(name = "CART_ID", unique = true)
+    private Long cartId;
+}
```
Comment (waleczna) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java:25]: Missing many to many relation to `Product` entity.

File: src/main/java/com/kodilla/ecommercee/domain/Order.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@Entity
+@Table(name = "ORDERS")
+public class Order {
+
+    @Id
+    @GeneratedValue
+    @NotNull
+    @Column(name = "ORDER_ID", unique = true)
+    private Long orderId;
+}
```
Comment (waleczna) [inline src/main/java/com/kodilla/ecommercee/domain/Order.java:22]: Missing many to many relation to `Product` entity.

**Review by waleczna — CHANGES_REQUESTED** [review]

File: src/main/java/com/kodilla/ecommercee/domain/Product.java (hunk 2 above)
Comment (CodeStudent1742) [reply src/main/java/com/kodilla/ecommercee/domain/Product.java]: Done

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 1 above)
Comment (CodeStudent1742) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: Done

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 3 above)
Comment (CodeStudent1742) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java:30]: done

**Review by BigKowalsky — APPROVED** [review]
In my opinion everything is ok. I analyzed changes since the last commit, In my IntelliJ gradle is building, app also starting without problems.

**Review by pietryczko — APPROVED** [review]
For me, everything is ok

**Pull request merged**